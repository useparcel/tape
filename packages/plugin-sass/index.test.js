import { tape } from "../tape/index.ts";
import { sass } from "./index.js";

describe("sass plugin", () => {
  test("should convert an external sass file to css", async () => {
    const results = await tape({
      entry: "/index.html",
      plugins: [sass],
      files: {
        "/index.html": {
          content: `
            <html>
              <head>
                <link rel="stylesheet" href="/style.scss">
              </head>
              <body>
                body
              </body>
            </html>
          `,
        },
        "/style.scss": {
          content: `
            $my-color: blue;

            body {
              background: $my-color;
            }
          `,
        },
      },
    });

    expect(results).toMatchSnapshot();
  });

  test("should convert embedded sass to css", async () => {
    const results = await tape({
      entry: "/index.html",
      plugins: [sass],
      files: {
        "/index.html": {
          content: `
            <html>
              <head>
                <style type="text/scss">
                  $my-color: blue;

                  body {
                    background: $my-color;
                  }
                </style>
              </head>
              <body>
                body
              </body>
            </html>
          `,
        },
      },
    });

    expect(results).toMatchSnapshot();
  });

  test("should throw an error with invalid sass", async () => {
    await expect(
      tape({
        entry: "/index.html",
        plugins: [sass],
        files: {
          "/index.html": {
            content: `
            <html>
              <head>
                <link rel="stylesheet" href="/style.scss">
              </head>
              <body>
                body
              </body>
            </html>
          `,
          },
          "/style.scss": {
            content: `
            $my-color: blue // missing a semi-colon

            body {
              background: $my-color;
            }
          `,
          },
        },
      })
    ).rejects.toThrow();
  });
});
