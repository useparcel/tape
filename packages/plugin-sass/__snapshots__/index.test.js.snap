// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sass plugin should convert an external sass file to css 1`] = `
Object {
  "diagnostics": Array [],
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
            <html>
              <head>
                <link rel=\\"stylesheet\\" href=\\"/style.css\\">
              </head>
              <body>
                body
              </body>
            </html>
          ",
    },
    "/style.css": Object {
      "content": "body {
  background: blue;
}",
    },
  },
}
`;

exports[`sass plugin should convert embedded sass to css 1`] = `
Object {
  "diagnostics": Array [],
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
            <html>
              <head>
                <style type=\\"text/scss\\">body {
  background: blue;
}</style>
              </head>
              <body>
                body
              </body>
            </html>
          ",
    },
  },
}
`;
